hcl2template/addrs/plugin.go

```diff
@@ -98,67 +134,63 @@ func IsPluginPartNormalized(str string) (bool, error) {
 //
 // The following are valid source string formats:
 //
-//	name
 //	namespace/name
 //	hostname/namespace/name
 func ParsePluginSourceString(str string) (*Plugin, hcl.Diagnostics) {
-	ret := &Plugin{
-		Hostname:  "",
-		Namespace: "",
-	}
 	var diags hcl.Diagnostics
 
-	// split the source string into individual components
-	parts := strings.Split(str, "/")
-	if len(parts) != 3 {
-		diags = diags.Append(&hcl.Diagnostic{
+	// Trim both leading and trailing slashes for convenience
```

Comment:
I suggest adding a call out on why we Trim,as apposed to calling out we trim for convenience. Looking at the parts function we want to make sure that the plugin source is in a valid addrs format and not a root file path. Is that correct?

Reply:
Yeah good call out. I used `Trim` here to drop the leading/trailing slashes so the `url.Parse` returns something expected in our case. Technically having a URL that starts with `//` or that ends with `/` is valid, but throws off our parsing, so I defensively remove them here to not have to deal with special cases like those later on.

Though I presume we could alternatively explicitely reject URIs formed like this, this might be clearer for users.